src/pedal/config.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{pedal, Pedal as PedalTrait, Result};
+use cpal::StreamConfig;
+use serde::Deserialize;
+use Pedal::*;
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pedals: Vec<Pedal>,
+}
+
+impl Config {
+    pub fn from(yaml: &str) -> Result<Config> {
+        Ok(serde_yaml::from_str(yaml)?)
+    }
+
+    pub fn default() -> Self {
+        Self {
+            pedals: vec![Transparent],
+        }
+    }
+
+    pub fn to_pedals(&self, stream_config: &StreamConfig) -> Result<Vec<pedal::Boxed>> {
+        self.pedals
+            .iter()
+            .map(|pedal| pedal.to_pedal(stream_config))
+            .collect()
```

Comment:
At the end of the chain, you `collect` back into a concrete collection type. It's using type inference to determine which type is needed here, but you could be explicit if necessary.

It's actually doing something pretty fancy here:`pedal.to_pedal(...)` produces a `Result<boxed::Pedal>`, so we end up with an iterable of `Result<boxed::Pedal>`, but we actually want a `Result<Vec<boxed::Pedal>>`, so it is moving the `Result` up a layer in the type. It'll be smart and stop iterating if any of the pedals result in an `Err`, and immediately bubble up that error.